### ISSUE TITLE
Compiler: Fatal error processing with Manifold type processor: manifold.ext.ExtensionManifold   on type: _temp_._Manifold_Temp_Main_

### ISSUE THREAD

**Issue by schittli** [manifold-systems/manifold#222]
**Describe the bug**
Every some day, building my java project stops with this exception (the stacktrace is below):
```
java: Fatal error processing with Manifold type processor: manifold.ext.ExtensionManifold
  on type: _temp_._Manifold_Temp_Main_1
  Please report the error with the accompanying stack trace.
  java.lang.NullPointerException
```

If I recompile the project afterwards, the exception disappears.

**To Reproduce**
Unfortunately, I can not reproduce the exception. 
I recompiled the project 20 times and I've never got the exception, therefore, I guess that it depends on the modified file relationship.
The only additional information is, that I _always_ get two strange `java.lang.NullPointerException` messages, too:

![image](https://user-images.githubusercontent.com/8282673/96137307-cf748700-0efc-11eb-8162-9d603f1ee55a.png)

**Expected behavior**
- Each exception should offer additional information, e.g. just a `java.lang.NullPointerException` message is strange.

**Screenshots**
Nothing else.

**Desktop (please complete the following information):**
- OS Type & Version: Windows 10, x64, 2008, Build 19041.338
- Java/JDK version: 1.8.0_162
- IDE version: IntelliJ 2020.2.3
- Manifold version: Maven: systems.manifold:manifold-all:2020.1.28
- Manifold IntelliJ plugin version: 2020.1.28

**Additional context**
-

**Stack trace**

```
java: Fatal error processing with Manifold type processor: manifold.ext.ExtensionManifold
  on type: _temp_._Manifold_Temp_Main_1
  Please report the error with the accompanying stack trace.
  java.lang.NullPointerException
  	at manifold.util.concurrent.ConcurrentWeakHashMap.get(ConcurrentWeakHashMap.java:1009)
  	at manifold.util.ReflectUtil.getRawMethodFromCache(ReflectUtil.java:838)
  	at manifold.util.ReflectUtil.getMethodFromCache(ReflectUtil.java:828)
  	at manifold.util.ReflectUtil.method(ReflectUtil.java:198)
  	at manifold.util.ReflectUtil.method(ReflectUtil.java:176)
  	at manifold.ext.ExtensionTransformer.getIncrementalCompileDrivers(ExtensionTransformer.java:1403)
  	at manifold.ext.ExtensionTransformer.findDrivers(ExtensionTransformer.java:1363)
  	at manifold.ext.ExtensionTransformer.incrementalCompile(ExtensionTransformer.java:1410)
  	at manifold.ext.ExtensionTransformer.incrementalCompileClasses(ExtensionTransformer.java:1346)
  	at manifold.ext.ExtensionTransformer.visitClassDef(ExtensionTransformer.java:1223)
  	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:693)
  	at manifold.ext.ExtensionManifold.process(ExtensionManifold.java:261)
  	at manifold.internal.javac.TypeProcessor.process(TypeProcessor.java:96)
  	at manifold.internal.javac.CompiledTypeProcessor.finished(CompiledTypeProcessor.java:251)
  	at com.sun.tools.javac.api.ClientCodeWrapper$WrappedTaskListener.finished(ClientCodeWrapper.java:681)
  	at com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:111)
  	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1342)
  	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1296)
  	at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
  	at com.sun.tools.javac.main.Main.compile(Main.java:523)
  	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
  	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
  	at org.jetbrains.jps.javac.JavacMain.compile(JavacMain.java:207)
  	at org.jetbrains.jps.incremental.java.JavaBuilder.compileJava(JavaBuilder.java:492)
  	at org.jetbrains.jps.incremental.java.JavaBuilder.compile(JavaBuilder.java:344)
  	at org.jetbrains.jps.incremental.java.JavaBuilder.doBuild(JavaBuilder.java:269)
  	at org.jetbrains.jps.incremental.java.JavaBuilder.build(JavaBuilder.java:222)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.runModuleLevelBuilders(IncProjectBuilder.java:1448)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuildersForChunk(IncProjectBuilder.java:1126)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.buildTargetsChunk(IncProjectBuilder.java:1193)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunkIfAffected(IncProjectBuilder.java:1087)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunks(IncProjectBuilder.java:853)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuild(IncProjectBuilder.java:443)
  	at org.jetbrains.jps.incremental.IncProjectBuilder.build(IncProjectBuilder.java:184)
  	at org.jetbrains.jps.cmdline.BuildRunner.runBuild(BuildRunner.java:135)
  	at org.jetbrains.jps.cmdline.BuildSession.runBuild(BuildSession.java:294)
  	at org.jetbrains.jps.cmdline.BuildSession.run(BuildSession.java:127)
  	at org.jetbrains.jps.cmdline.BuildMain$MyMessageHandler.lambda$channelRead0$0(BuildMain.java:218)
  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  	at java.lang.Thread.run(Thread.java:748)
```

**Comment by rsmckinney** [comment]
This is a difficult bug to pin down. It is limited to IntelliJ builds (JPS) and appears to be a timing issue with file system writes; the _temp_._Manifold_Temp_Main_1 files are a transient artifact of the plugin's interface with JPS and should not exist after a build completes.

**Comment by rsmckinney** [comment]
fixed with manifold release 2020.1.31 + manifold ij plugin releases 2019.3.31, 2020.1.31, 2020.3.31
